scripts/build-windows.ps1

```diff
@@ -0,0 +1,175 @@
+   
+    if ($installCmake -eq "y" -or $installCmake -eq "yes") {
+        # download cmake 3.17.3 64-bit and run MSI for user
+        $client = New-Object System.Net.WebClient        
+        $tempDownloadFile = New-TemporaryFile
+        $cmakeMsiFile = "$tempDownloadFile.cmake-3.17.3-win64-x64.msi"
+        Write-Output "Downloading cmake (temporary file location $cmakeMsiFile)"
+        Write-Output "Note: select the option to add cmake to path for this script to operate"
+        $client.DownloadFile("https://github.com/Kitware/CMake/releases/download/v3.17.3/cmake-3.17.3-win64-x64.msi", "$cmakeMsiFile")
+        Start-Process $cmakeMsiFile -Wait
+        Remove-Item $cmakeMsiFile
+        Write-Output "Cmake should have installed, this script will now exit because of path updates - please now re-run this script"
+        exit
+    }
+    else{
+        Write-Output "Quitting because cmake is required"     
+        exit
+    }
+}
+
+if ( $CXX11 -eq "OFF" ) {
+    # get pthreads (this is legacy, and is only availble in nuget for VS2015 and VS2013)
+    if ( $VS_VERSION -gt 2015 ) { 
+        Write-Output "Pthreads is not recommended for use beyond VS2015 and is not supported by this build script - aborting build"
+        exit
+    }
+    if ( $DEVENV_PLATFORM -eq 'Win32' ) { 
+        nuget install cinegy.pthreads-win32-$VS_VERSION -version 2.9.1.24 -OutputDirectory ../_packages
+    }
+    else {        
+        nuget install cinegy.pthreads-win64-$VS_VERSION -version 2.9.1.24 -OutputDirectory ../_packages
+    }
+}
+
+if ($STATIC_LINK_SSL -eq "ON") {
+    # requesting a static link will implicitly enable encryption support
+    Write-Output "Static linking to OpenSSL requested, will force encryption feature ON"
+    $ENABLE_ENCRYPTION = "ON"
+}
+
+# build the cmake command flags from arguments
+$cmakeFlags = "-DCMAKE_BUILD_TYPE=$CONFIGURATION " + 
+                "-DENABLE_STDCXX_SYNC=$CXX11 " + 
+                "-DENABLE_APPS=$BUILD_APPS " + 
+                "-DENABLE_ENCRYPTION=$ENABLE_ENCRYPTION " +
+                "-DOPENSSL_USE_STATIC_LIBS=$STATIC_LINK_SSL " + 
+                "-DENABLE_UNITTESTS=$UNIT_TESTS"
+
+# cmake uses a flag for architecture from vs2019, so add that as a suffix
+if ( $VS_VERSION -eq '2019' ) {    
+    $cmakeFlags += " -A `"$DEVENV_PLATFORM`""
+}
+
+# fire cmake to build project files
+$execVar = "cmake ../ -G`"$CMAKE_GENERATOR`" $cmakeFlags"
+Write-Output $execVar
+Invoke-Expression "& $execVar"
+
+# check build ran OK, exit if cmake failed
+if($LASTEXITCODE -ne 0) {
+    return $LASTEXITCODE
+}
+
+# run the set-version-metadata script to inject build numbers into appveyors console and the resulting DLL
+. $PSScriptRoot/set-version-metadata.ps1
+
+# execute msbuild to perform compilation and linking - if outside of appveyor, use vswhere to find msbuild
+if($Env:APPVEYOR) { 
+    msbuild SRT.sln /p:Configuration=$CONFIGURATION /p:Platform=$DEVENV_PLATFORM
```

Comment:
This command should be saved in the variable and executed at the end of a procedure of finding one. Please do not multiply instructions containing the same data (command-line options in this case). The command defined here should be executed in line 160, this time outside the conditional block.

Reply:
Yeah, this came about because of a) multiple re-writes during sorting this out jumping between machines that had different environments and b) trying to extend what was in the original script without wholesale re-writing (that part above was inherited from the original appveyor code).

I actually spent a lot of time avoiding falling into exactly that trap in the cmake invocation (because of the annoying change in cmake to pass architecture as a discreet flag for VS2019+), and then never even noticed I'd repeated it here.

Will change it.

Reply:
That's what peer reviews are intended for :)